pandas-stubs/_libs/tslibs/timestamps.pyi

```diff
@@ -301,3 +301,6 @@ class Timestamp(datetime):
     def days_in_month(self) -> int: ...
     @property
     def daysinmonth(self) -> int: ...
+    @property
+    def unit(self) -> str: ...
+    def as_unit(self, unit: str, round_ok: bool = ...) -> Timestamp: ...
```

Comment:
The only possible values for the return of `unit` are `Literal["s", "ms", "us", "ns"]`  In pandas, this is now defined in `pandas._typing.py` via:
```python
TimeUnit = Literal["s", "ms", "us", "ns"]
```
So can you do the following:
1. Add that line to `pandas-stubs/_typing.pyi` 
2. Make the return value of `unit` be `TimeUnit`
3. Make the allowable values of `unit` in `as_unit` to be `TimeUnit`

You probably will have to change the test for `ts.unit` to use `assert_type()` testing against `TimeUnit`

If you want to go beyond that, the declaration `Literal["s", "ms", "us", "ns"]` appears in the following files and should be replaced with `TimeUnit`:

- pandas-stubs/core/frame.pyi
- pandas-stubs/core/indexes/datetimes.pyi
- pandas-stubs/core/series.pyi
- pandas-stubs/io/json/_json.pyi

But if you don't want to make that change, it's OK - I can just create a new issue for that once this PR is done.